### PR TITLE
docs(release): add v1.4.0 release notes

### PR DESCRIPTION
**Pull request by jpadilla**
## Description

Add `v1.4.0` release details to CHANGELOG

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,13 @@ Try to keep listed changes to a concise bulleted list of simple explanations of
 
 ## [Unreleased]
 
+## [v1.4.0] - 2023-12-08
```
Comment (jon-whit) [inline CHANGELOG.md:11]: Be sure to add the link at the bottom of this file.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,13 @@ Try to keep listed changes to a concise bulleted list of simple explanations of
...
+
+[Full changelog](https://github.com/openfga/openfga/compare/v1.3.10...v1.4.0)
+
+### Changed
+* Enable support for ABAC Conditional Relationships by default. ([#1220](https://github.com/openfga/openfga/pull/1220))
```
Comment (aaguiarz) [inline CHANGELOG.md:16]: We are not calling it ABAC in the docs, just "Conditional Relationships"

Comment (jon-whit) [reply CHANGELOG.md:16]: ABAC Conditions + Conditional Relationship Tuples

Those are the terms we're using

Comment (jon-whit) [reply CHANGELOG.md:16]: https://github.com/openfga/openfga.dev/pull/583

**Review by jon-whit — APPROVED** [review]

**Review by miparnisari — APPROVED** [review]

**Review by adriantam — APPROVED** [review]

**Pull request merged**